File: leetcode-submissions/102_Binary_Tree_Level_Order_Traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:

    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>>ans;
        if(root==NULL) return ans;
        queue<TreeNode*>q;
        q.push(root);
        while(!q.empty()){
            vector<int>level;
            int currsize=q.size();
            while(currsize--){
                TreeNode* currnode=q.front();
                cout<<currnode->val<<endl;
                q.pop();
                level.push_back(currnode->val);
                if(currnode->left!=NULL){
                    q.push(currnode->left);
                }
                if(currnode->right!=NULL){
                    q.push(currnode->right);
                }
            }
            ans.push_back(level);
                
        }
        
    return ans;
    }
};

File: leetcode-submissions/102_Binary_Tree_Level_Order_Traversal.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function levelOrder(root: TreeNode | null): number[][] {
    if(! root){
        return [];
    }
    const queue:TreeNode[]= [root];
    const ans:number[][]= [];
    while(queue.length>0){
        let size = queue.length;
        const level:number[]= [];
        while(size--){
            const currNode:TreeNode = queue[0];
            level.push(currNode.val);
            queue.shift();
            if(currNode.left){
                queue.push(currNode.left);
            }
            if(currNode.right){
                queue.push(currNode.right);
            }
            
        }
        ans.push(level);
        
    }
    return ans ;
};